src/SFA.DAS.EmployerIncentives/Infrastructure/Api/PassThroughApiClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerIncentives.Interfaces;
+
+namespace SFA.DAS.EmployerIncentives.Infrastructure.Api
+{
+    public class PassThroughApiClient : IPassThroughApiClient
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<PassThroughApiClient> _logger;
+
+        public PassThroughApiClient(HttpClient httpClient, ILogger<PassThroughApiClient> logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        public async Task<InnerApiResponse> GetAsync(Uri uri, object queryData = null, CancellationToken cancellationToken = default)
+        {
+            if (queryData != null)
+            {
+                uri = new Uri(AddQueryString(uri.ToString(), queryData), UriKind.RelativeOrAbsolute);
+            }
+
+            var response = await _httpClient.GetAsync(uri, cancellationToken).ConfigureAwait(false);
+            return await CreateApiResponse(response);
+        }
+
+        public Task<InnerApiResponse> GetAsync(string uri, object queryData = null, CancellationToken cancellationToken = default)
+        {
+            return GetAsync(new Uri(uri, UriKind.RelativeOrAbsolute), queryData, cancellationToken);
+        }
+
+        public Task<InnerApiResponse> PostAsync(string uri, CancellationToken cancellationToken = default)
+        {
+            return PostAsync<object>(uri, null, cancellationToken);
+        }
+
+        public async Task<InnerApiResponse> PostAsync<TRequest>(string uri, TRequest request, CancellationToken cancellationToken = default) where TRequest : class
+        {
+            var response = await _httpClient.PostAsJsonAsync(uri, request, cancellationToken).ConfigureAwait(false);
+            return await CreateApiResponse(response);
+        }
+
+        public async Task<InnerApiResponse> DeleteAsync(string uri, CancellationToken cancellationToken = default)
+        {
+            var response = await _httpClient.DeleteAsync(uri, cancellationToken).ConfigureAwait(false);
+            return await CreateApiResponse(response);
+        }
+
+        private async Task<InnerApiResponse> CreateApiResponse(HttpResponseMessage responseMessage)
```

Comment:
Do we need this?  Is there a reason not to just return the HttpResponse?

Reply:
I agree - I think I'm also struggling with the concept of this class - are you using everything that comes back from the response in the front end? 
For `POST` and `DELETE` I'm unsure as to why you can't use the other client?

Reply:
The intent of the class is to keep it as light as possible. I worry about returning the HttpResponse from this class as it's content is a stream and I'm unsure what happens to it if when the client goes out of scope.

I don't want to use the other client as this means we have to start putting more infrastructure class in place to deal with the different types of responses. Such as validation messages